storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java

```diff
@@ -70,18 +70,20 @@ public TopicBasedRemoteLogMetadataManager topicBasedRlmm() {
     }
 
     @Test
-    public void testInternalTopicExists() {
+    public void testDoesTopicExist() {
         Properties adminConfig = remoteLogMetadataManagerHarness.adminClientConfig();
         ListenerName listenerName = remoteLogMetadataManagerHarness.listenerName();
         try (Admin admin = remoteLogMetadataManagerHarness.createAdminClient(listenerName, adminConfig)) {
-            String topic = topicBasedRlmm().config().remoteLogMetadataTopicName();
+            String topic = "test-topic-exist";
+            remoteLogMetadataManagerHarness.createTopic(topic, 1, 1, new Properties(),
```

Comment:
`createTopic` only waits for the topic metadata to be committed in the metadata log, but not waiting for the metadata to be propagated to every broker. `topicBasedRlmm().doesTopicExist` calls `describeTopic` and only checks the metadata in a least loaded broker. So, it seems that there is no strong guarantee that `doesTopicExist` is always true?

Reply:
[TestUtils#createTopicWithAdmin](https://sourcegraph.com/github.com/apache/kafka@fade3d10ea07eea5d6076b8fb1b68e2db5ffec48/-/blob/core/src/test/scala/unit/kafka/utils/TestUtils.scala?L497) waits for the metadata to be propagated to all the brokers before returning the call so the test will pass. Also, tried running the test repeatedly 100 times to ensure that the test is not flaky.

Reply:
Yes, you're right, we'll make sure the metadata is propagated to all nodes for both KRaft and ZK modes.

Reply:
Thanks for the explanation, @kamalcph. The PR LGTM then.